### PR TITLE
Add a new HAFS regression test with Thompson MP and GFDL surface layer schemes and Corrected S4 template regression testing job card

### LINKED ISSUE #900: Turn on the four HAFS regional/global nesting RTs on Jet
## Description
The four HAFS regional/global nesting RTs run too slow than on other platforms, thus were not enabled running on Jet.

## Solution
Need to look into and fix this so that we can turn on the HAFS regional/global nesting RTs on Jet.

## Alternatives
Currently, these four HAFS regional/global nesting RTs were turned off Jet.

## Related to
- associated with ufs-community/ufs-weather-model/pull/<pr_number>

### LINKED ISSUE #987: Add a new regression test for HAFS with the legacy of the HWRF suite
## Description
Since the HWRF physics suite was removed from the standard regression tests, we'd like to keep testing and potentially use one or two physics schemes from the legacy of the HWRF physics suite for the development of the future HAFS physics suite. 

## Solution
We constructed a new HAFS physics suite by replacing the GFS surface layer scheme with the GFDL surface layer scheme (a legacy scheme from the HWRF physics suite) and the GFDL microphysics scheme with the Thompson microphysics scheme.

## Alternatives
Keep the GFDL surface layer scheme in the repository if the new regression test is not approved.

### LINKED ISSUE #991: Broken template regression testing job card for S4
## Description
The template job card for S4, [fv3_slurm.IN_s4](https://github.com/ufs-community/ufs-weather-model/blob/58ab0e27f582d695636a3c730adf8f21a1293c3b/tests/fv3_conf/fv3_slurm.IN_s4#L17), has spaces in the assignment `MACHINE_ID = s4`, which is invalid syntax.  This causes `rt.sh` to fail when attempting to run any test as the machine cannot be determined.

### PR DESCRIPTION
**Pull request by ChunxiZhang-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

Created a new regression test for the newly added HAFS physics suite, e.g., suite_FV3_HAFS_v0_thompson_tedmf_gfdlsf.xml (Note that the other newly added suite, e.g. suite_FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst.xml is not included in this regression test because this one is very similar to the first one). 

Corrected S4 template regression testing job card

Contributors: @ChunxiZhang-NOAA, @BinLiu-NOAA, @DavidHuber-NOAA

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes https://github.com/ufs-community/ufs-weather-model/issues/987
- fixes https://github.com/ufs-community/ufs-weather-model/issues/900
- fixes https://github.com/ufs-community/ufs-weather-model/issues/991

## Testing

Regression tests related to the HAFS were conducted on Hera.intel. All tests were passed except the newly added 'hafs_regional_atm_thompson_gfdlsf' due to missing baseline.

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/fv3atm/pull/459

### CODE REVIEW

File: tests/parm/field_table/field_table_hafs_thompson [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# added by FRE: sphum must be present in atmos
```
Comment (climbfuji) [inline tests/parm/field_table/field_table_hafs_thompson:1]: What is the difference between this file and `https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/field_table/field_table_thompson_aero_tke`?

File: tests/rt.conf [hunk 2]
```diff
@@ -142,8 +142,9 @@ RUN     | control_wam_debug
 # HAFS tests                                                                                                                                                                      #
 ###################################################################################################################################################################################
 #
-COMPILE | -DAPP=HAFSW -DCCPP_SUITES=FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_gfdlmp_tedmf_nonsst -D32BIT=ON                           | - wcoss2                                | fv3 |
+COMPILE | -DAPP=HAFSW -DCCPP_SUITES=FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_thompson_tedmf_gfdlsf,FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst,FV3_HAFS_v0_gfdlmp_tedmf_nonsst -D32BIT=ON                           | - wcoss2                                | fv3 |
```
Comment (BinLiu-NOAA) [inline tests/rt.conf]: @ChunxiZhang-NOAA, could you change this line into the following? Thanks!
COMPILE | -DAPP=HAFSW -DCCPP_SUITES=**FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_gfdlmp_tedmf_nonsst,FV3_HAFS_v0_thompson_tedmf_gfdlsf,FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst** -D32BIT=ON                           | - wcoss2                                | fv3 |

File: tests/tests/hafs_regional_atm_thompson_gfdlsf [hunk 3]
```diff
@@ -0,0 +1,98 @@
+###############################################################################
+#
+#  HAFS regional atmosphere only
```
Comment (BinLiu-NOAA) [inline tests/tests/hafs_regional_atm_thompson_gfdlsf]: @ChunxiZhang-NOAA, Please change this line into:
HAFS regional atmosphere only using the Thompson microphysics and GFDL surface layer schemes
or something similar.

File: tests/tests/hafs_regional_atm_thompson_gfdlsf [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+#
+###############################################################################
+
+export TEST_DESCR="Compare HAFS regional atmosphere only results with previous trunk version"
```
Comment (BinLiu-NOAA) [inline tests/tests/hafs_regional_atm_thompson_gfdlsf]: @ChunxiZhang-NOAA, also please update this line into the following or something similar. Thanks!
export TEST_DESCR="Compare HAFS regional atmosphere only results using the Thompson microphysics and GFDL surface layer schemes with previous trunk version"

File: tests/parm/field_table/field_table_hafs_thompson (hunk 1 above)
Comment (ChunxiZhang-NOAA) [reply tests/parm/field_table/field_table_hafs_thompson:1]: field_table_hafs_thompson is the same as field_table_thompson_noaero_tke. I just want to hold a spot for future changes of field_table_hafs_thompson. Note that field_table_hafs is the same as field_table_gfsv16.

File: tests/rt.conf (hunk 2 above)
Comment (ChunxiZhang-NOAA) [reply tests/rt.conf]: Changed as you suggested.

File: tests/tests/hafs_regional_atm_thompson_gfdlsf (hunk 3 above)
Comment (ChunxiZhang-NOAA) [reply tests/tests/hafs_regional_atm_thompson_gfdlsf]: Changed as you suggested

File: tests/tests/hafs_regional_atm_thompson_gfdlsf (hunk 4 above)
Comment (ChunxiZhang-NOAA) [reply tests/tests/hafs_regional_atm_thompson_gfdlsf]: Changed to "Compare the results from HAFS regional atmosphere only using the Thompson microphysics scheme and GFDL surface layer scheme with previous trunk version".

**Comment by binli2337** [comment]
@ChunxiZhang-NOAA Please update your PR and add the changes from PR #992. Thanks.

**Comment by ChunxiZhang-NOAA** [comment]
I combined David's change in PR#992 to this PR branch and synced my branch with the latest develop branch for both FV3 and ufs-weather-model level.  New baseline for this 'hafs_regional_atm_thompson_gfdlsf' regression test need to be added.

**Review by BinLiu-NOAA — APPROVED** [review]
Changes look good to me.

File: tests/parm/input_regional_hafs.nml.IN [hunk 5]
```diff
@@ -158,7 +158,12 @@
        fhcyc          = @[FHCYC]
        use_ufo        = .true.
        pre_rad        = .false.
-       imp_physics    = 11
+       imp_physics    = @[IMP_PHYSICS]
+       ltaerosol      = .false.
+       lradar         = .true.
+       ttendlim       = 0.004
+       sedi_semi      = .false.
+       ext_diag_thompson = .false.
```
Comment (junwang-noaa) [inline tests/parm/input_regional_hafs.nml.IN:166]: Just to confirm those variables ltaerosol, lradar, ttendlim and sedi_semi are only used by Thompson, they have no impact on other MP schemes, right?

**Comment by DavidHuber-NOAA** [comment]
Looks good to me, also.  Thanks @ChunxiZhang-NOAA!

File: tests/parm/input_regional_hafs.nml.IN (hunk 5 above)
Comment (ChunxiZhang-NOAA) [reply tests/parm/input_regional_hafs.nml.IN:166]: Confirmed: those variables ltaerosol, lradar, ttendlim and sedi_semi are only used by Thompson, and they have no impact on other MP schemes.

**Comment by binli2337** [comment]
@ChunxiZhang-NOAA The updated code cannot be compiled on Hera and Gaea.

**Comment by binli2337** [comment]
@ChunxiZhang-NOAA Please see error messages at /scratch1/NCEPDEV/stmp2/Bin.Li/FV3_RT/rt_19027/compile_001.

**Comment by ChunxiZhang-NOAA** [comment]
Hi Bin, sorry for that. I found it is caused by a suite we added, e.g., FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst. The regression test we added actually didn't use this suite. I will delete this suite and push the changes to my fork.

**Review by climbfuji — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/814264612/20220118194511/ufs-weather-model
/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220113/GNU
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
cheyenne-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/814264612/20220118194612/ufs-weather-model
/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220113/INTEL
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
cheyenne-intel-BL

**Comment by climbfuji** [comment]
Did you just want to verify against the baseline? It says 20220113 already exists.

**Comment by junwang-noaa** [comment]
We just need to add one test hafs_regional_atm_thompson_gfdlsf's results to current baseline.

**Comment by binli2337** [comment]
@ChunxiZhang-NOAA @junwang-noaa All tests passed on Cray. RegressionTests_wcoss_cray.log is located at /gpfs/hps3/stmp/Bin.Li/ufs-weather-model/tests.

**Review by junwang-noaa — APPROVED** [review]

**Review by binli2337 — APPROVED** [review]

**Pull request merged**